parser/prism/Translator.cc

```diff
@@ -3032,24 +3045,72 @@ unique_ptr<parser::Regopt> Translator::translateRegexpOptions(pm_location_t clos
         options = string_view();
     }
 
-    return make_unique<parser::Regopt>(translateLoc(closingLoc), options);
+    if (!directlyDesugar) {
+        return make_unique<parser::Regopt>(location, options);
+    }
+
+    // Desugar options to integer flags
+    int flags = 0;
+    for (auto &chr : options) {
+        switch (chr) {
+            case 'i':
+                flags |= 1; // Regexp::IGNORECASE
+                break;
+            case 'x':
+                flags |= 2; // Regexp::EXTENDED
+                break;
+            case 'm':
+                flags |= 4; // Regexp::MULTILINE
+                break;
+            default:
+                // Encoding options (n, e, s, u) are handled by the parser
```

Comment:
Not pressing for this PR, but we should confirm that these 4 assumptions hold for Prism, as well.

Can you open an issue for it?